Add alert message to confirm success on patient form submission.

takidisk opened the pull request:
#### Short description of what this resolves:
This pull request adds a simple alert message when submitting the new patient encounter form to provide basic user feedback.

#### Changes proposed in this pull request:
- Created a `showSuccess()` JavaScript function in `common.html.twig`.
- Added `onsubmit="return showSuccess();"` to the `<form>` element.

adunsulag commented on interface/forms/newpatient/templates/newpatient/common.html.twig:
```diff
@@ -15,6 +15,13 @@
 <!DOCTYPE html>
 <html>
+<script>
+    function showSuccess(){
```
You may not be aware but all the scripts for this form should go in the newpatient.js file.  You should add the onSubmit via the javascript addEventListener piece.
https://github.com/openemr/openemr/blob/3a1343e3861caf760d199e8385f17f26419d5d13/interface/forms/newpatient/newpatient.js#L54-L53

We did a lot of work to take the javascript stuff out of the twig file and we'd like to keep it that way.

You'l also want to use window.top.xlj in order to get language translation in the javascript.  

Finally bump up the javascript asset version when you make that change so that people get the latest js file.
https://github.com/openemr/openemr/blob/08ed79b7bc8f60f7dec52fa9fb9d93240a5c2a39/version.php#L50

Thank you for the PR and feel free to ask questions if you need help.

adunsulag commented:
Forgot to mention that in order for our changelog generator to work please open a github issue and attach the issue you fixed in your PR description.  IE: "Fixes #issue-number".

takidisk commented:
Thank you for the feedback! 

I have now: 
- Moved the JavaScript logic into `newpatient.js` using `addEventListener`, 
- Used `xl()` for the translation support, 
- Removed the inline JS from `common.html.twig`,
- Bumped `$v_js_includes` to `79` in `version.php`, 
- Opened and linked issue [#8379] (https://github.com/openemr/openemr/issues/8379) as requested.

Let me know if anything else needs to be adjusted.

mdsupport commented on interface/forms/newpatient/newpatient.js:
```diff
@@ -191,6 +191,14 @@ window.NewPatientForm = function(window) {
+        const form = document.getElementById('new-encounter-form');
+        if (form) {
+            form.addEventListener('submit', function() {
+                alert(xl('Form submitted successfully!'));
```
alerts force a click when staff is juggling many tasks. If a feedback is must, consider bootstrap's of alert classes. Unlike js alert box, you can even include other html elements to make it part of a workflow. Add an timer and the alert can optionally close itself after n seconds.

takidisk commented:
I replaced the `alert()` with a Bootstrap dismissible alert as suggested and ensured that `bootstrap.bundle.min.js` is loaded. 
Thanks again for the feedback!

stephenwaite commented:
looking good @takidisk!

here's some git tips:
https://www.open-emr.org/wiki/index.php/Git_for_dummies#Tenet_1:_Branches_are_Supreme
https://github.com/openemr/openemr/pull/657#issuecomment-295630669

takidisk commented:
Hi @mdsupport @sjpadgett @adunsulag 👋

Just checking in again — this PR has been marked as "Needs Review" for a while, and I haven’t received a response since my last follow-up two weeks ago. All requested changes have been addressed, including:

- Replacing `alert()` with a Bootstrap dismissible alert
- Loading `bootstrap.bundle.min.js` if needed
- Moving JS to `newpatient.js`
- Using `xl()` for translations
- Bumping `v_js_includes` to force refresh

Please let me know if there’s anything further needed on my end — happy to make adjustments. Thanks again for your time and guidance!

mdsupport commented:
@takidisk, Sorry for the delay. Project admins decide if (and when) to approve changes. Thank you for followup.

sjpadgett reviewed (commented):
Sorry, I couldn't get to work! Anyone else test?
